app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt

```diff
@@ -25,6 +32,10 @@ class LoginViewModel(app: Application) : AndroidViewModel(app) {
     val validEmail: LiveData<Boolean>
         get() = _validEmail
 
+    private val _response = MutableLiveData<String>()
```

Comment:
The response isn't a String. The response should be an UserInfo object. This is what userRepository.getUserInfo(userAuth).body() holds. Also, a variable name "response" is very general. Always try to think a name for exactly what it is (it's not simply a response, moreover, it's the body of a response that represent something else in your domain)